pkgs/development/libraries/rapidfuzz-cpp/default.nix

```diff
@@ -33,7 +33,7 @@ stdenv.mkDerivation rec {
     catch2_3
   ];
 
-  doCheck = true;
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
```

Comment:
this is the default https://github.com/NixOS/nixpkgs/blob/cf4b4ea59039b689f11413dccacff7b19a6e7e91/pkgs/stdenv/generic/make-derivation.nix#L171

Reply:
oh, the final value of `doCheck` isn't visible at this point in the derivation, so the above `cmakeFlags = lib.optionals doCheck ...` wouldn't evaluate as desired.

here's an alternate way to achieve the same thing. i'll let you decide if it's better or not (feel free to commit it if you think so):
```diff
diff --git a/pkgs/development/libraries/rapidfuzz-cpp/default.nix b/pkgs/development/libraries/rapidfuzz-cpp/default.nix
index 9cde99b69f0..04480a5d31f 100644
--- a/pkgs/development/libraries/rapidfuzz-cpp/default.nix
+++ b/pkgs/development/libraries/rapidfuzz-cpp/default.nix
@@ -20,9 +20,11 @@ stdenv.mkDerivation rec {
     cmake
   ];
 
-  cmakeFlags = lib.optionals doCheck [
-    "-DRAPIDFUZZ_BUILD_TESTING=ON"
-  ];
+  preConfigure = ''
+    if [ -n "$doCheck" ]; then
+      cmakeFlags="$cmakeFlags -DRAPIDFUZZ_BUILD_TESTING=ON"
+    fi
+ '';
 
   CXXFLAGS = lib.optionals stdenv.cc.isClang [
     # error: no member named 'fill' in namespace 'std'
@@ -33,8 +35,6 @@ stdenv.mkDerivation rec {
     catch2_3
   ];
 
-  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
-
   meta = {
     description = "Rapid fuzzy string matching in C++ using the Levenshtein Distance";
     homepage = "https://github.com/maxbachmann/rapidfuzz-cpp";
```

i may peek upstream to see why they're using `RAPIDFUZZ_BUILD_TESTING` instead of `BUILD_TESTING`: might be able to remove this stuff here if they want to switch that.

Reply:
the proper way is to access `finalAttrs.doCheck` like here https://github.com/NixOS/nixpkgs/blob/d682cbf1548727b57ed1071e02399a4109384bb1/pkgs/development/libraries/nlohmann_json/default.nix